src/iop/highlights.c

```diff
@@ -531,55 +531,110 @@ static void process_lch_bayer(dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *pie
   }
 }
 
-static void process_lch_xtrans(
-    const void *const ivoid,
-    void *const ovoid,
-    const int width,
-    const int height,
-    const float clip,
-    const dt_iop_roi_t *const roi_in,
-    const uint8_t (*const xtrans)[6])
+static void process_lch_xtrans(dt_iop_module_t *self, const void *const ivoid,
+                               void *const ovoid, const dt_iop_roi_t *const roi_in,
+                               const dt_iop_roi_t *const roi_out, const float clip)
 {
+  const uint8_t (*const xtrans)[6] = self->dev->image_storage.xtrans;
+
 #ifdef _OPENMP
 #pragma omp parallel for schedule(static) default(none)
```

Comment:
Indeed, this must be the case. As some columns require way more work (have more highlights to reconstruct) it makes sense that with dynamic scheduling there was less chance of threads on non-clipped columns stalling out.